lib/registry.js

```diff
@@ -361,7 +356,7 @@ Registry.prototype.createDataSource = function(name, options) {
   };
 
   if (ds.settings && ds.settings.defaultForType) {
-    this.setDefaultDataSourceForType(ds.settings.defaultForType, ds);
+    console.warn('DataSource option "defaultForType" is no longer supported');
```

Comment:
The setting will be ignored. If there are any models depending on `autoAttach`, then these models will return an error whenever a DAO method is invoked, something like "The model is not attached to any datasource".

On the second thought, I agree with you it's better to throw an error here, so that the problem is detected as soon as possible (at startup). Will rework.